api/crosscontroller/crosscontroller.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package crosscontroller
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	names "gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	apiwatcher "github.com/juju/juju/api/watcher"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/crossmodel"
+	"github.com/juju/juju/watcher"
+)
+
+var logger = loggo.GetLogger("juju.api.crosscontroller")
+
+// Client provides access to the CrossController API facade.
+type Client struct {
+	base.ClientFacade
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client-side CrossModelRelations facade.
+func NewClient(caller base.APICallCloser) *Client {
+	frontend, backend := base.NewClientFacade(caller, "CrossController")
+	return &Client{
+		ClientFacade: frontend,
+		facade:       backend,
+	}
+}
+
+func (c *Client) ControllerInfo(controllerUUID string) (*crossmodel.ControllerInfo, error) {
```

Comment:
Given there's several of these now duplicated, is it time to introduce an api/common struct and embed in several api clients?

Reply:
No, because

the one I added to apiserver/crossmodelrelations shouldn't be there, I added it and forgot to drop it
this one and the one in crosscontroller aren't duplicates, they have different signatures